### PR TITLE
Add new platform azure support for testing

### PR DESCRIPTION
**Pull request by crombus**

### CODE REVIEW

File: ocs_ci/deployment/azure.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class AZUREUPI:
+    """
+    A class to handle Bare metal UPI specific deployment
```
Comment (sabose) [inline ocs_ci/deployment/azure.py:8]: change the comment - handle Azure IPI deployments

File: ocs_ci/ocs/platform_nodes.py [hunk 2]
```diff
@@ -32,7 +32,7 @@ class PlatformNodesFactory:
 
     """
     def __init__(self):
-        self.cls_map = {'AWS': AWSNodes, 'vsphere': VMWareNodes, 'aws': AWSNodes, 'baremetal': NodesBase}
+        self.cls_map = {'AWS': AWSNodes, 'azure': AZURENodes 'vsphere': VMWareNodes, 'aws': AWSNodes, 'baremetal': NodesBase}
```
Comment (sabose) [inline ocs_ci/ocs/platform_nodes.py:35]: missing comma

Comment (sabose) [reply ocs_ci/ocs/platform_nodes.py:35]: Also done't you need a class for AZURENodes?

Comment (prsurve) [inline ocs_ci/ocs/platform_nodes.py:35]: comma "," missing and azure should be using NodeBase class, as AZURENodes class is missing

File: ocs_ci/deployment/azure.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class AZUREUPI:
```
Comment (prsurve) [inline ocs_ci/deployment/azure.py:6]: AZUREIPI

File: conf/deployment/azure/ipi_1az_rhel7_3m_3w.yaml [hunk 4]
```diff
@@ -0,0 +1,7 @@
+---
+ENV_DATA:
+  platform: 'azure'
+  deployment_type: 'ipi'
+  worker_replicas: 3
+  master_replicas: 3
+  worker_instance_type: 'Standard_D16s_v3'
```
Comment (prsurve) [inline conf/deployment/azure/ipi_1az_rhel7_3m_3w.yaml:7]: this YAML is should be for RHCOS

File: ocs_ci/ocs/platform_nodes.py (hunk 2 above)
Comment (crombus) [reply ocs_ci/ocs/platform_nodes.py]: I added the NodesBase class for this.

File: conf/deployment/azure/ipi_1az_rhcos_3m_3w.yaml [hunk 5]
```diff
@@ -0,0 +1,7 @@
+---
+ENV_DATA:
+  platform: 'azure'
+  deployment_type: 'ipi'
+  worker_replicas: 3
+  master_replicas: 3
+  worker_instance_type: 'Standard_D16s_v3'
```
Comment (ebenahar) [inline conf/deployment/azure/ipi_1az_rhcos_3m_3w.yaml:7]: should we also have master instance type specified here?

File: conf/deployment/azure/ipi_1az_rhcos_3m_3w.yaml [hunk 6]
```diff
@@ -0,0 +1,8 @@
+---
+ENV_DATA:
+  platform: 'azure'
+  deployment_type: 'ipi'
+  worker_replicas: 3
+  master_replicas: 3
+  master_instance_type: 'Standard_D16s_v3'
```
Comment (ebenahar) [inline conf/deployment/azure/ipi_1az_rhcos_3m_3w.yaml:7]: need to change to the relevant master instance type

File: ocs_ci/ocs/platform_nodes.py [hunk 7]
```diff
@@ -32,7 +32,7 @@ class PlatformNodesFactory:
 
     """
     def __init__(self):
-        self.cls_map = {'AWS': AWSNodes, 'vsphere': VMWareNodes, 'aws': AWSNodes, 'baremetal': NodesBase}
+        self.cls_map = {'AWS': AWSNodes, 'azure': NodesBase ,'vsphere': VMWareNodes, 'aws': AWSNodes, 'baremetal': NodesBase}
```
Comment (prsurve) [inline ocs_ci/ocs/platform_nodes.py:35]: space required of ,

**Comment by mbukatov** [comment]
To unblock my work, I'm going to rebase this, fix flake errors and have this merged. This way, both work drafted in https://github.com/red-hat-storage/ocs-ci/pull/2325/files and my azure deployment work can continue without conflicts.

File: ocs_ci/deployment/azure.py [hunk 8]
```diff
@@ -0,0 +1,13 @@
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class AZUREIPI:
```
Comment (petr-balogh) [inline ocs_ci/deployment/azure.py:6]: You are failing without this definition mainly cause we are trying to instantiate deployer class even for testing.

I am fixing this issue here:  https://github.com/red-hat-storage/ocs-ci/pull/2307/files#diff-484462fced51d1a06b1d93b4a44dd535R1048  . But if we will do deployment here It's OK to have this class already defined and the work you done in the map...

**Pull request merged**